app/helpers/search_helper.rb

```diff
@@ -2,41 +2,34 @@ module SearchHelper
 
   # modified from mislav-will_paginate-2.3.2/lib/will_paginate/view_helpers.rb
   def search_header(collection, search, item_name, parent=nil)
-    header = ""
+    header = []
     if !collection.respond_to?(:total_pages)
-      header = ts "Recent #{item_name.pluralize}"
+      header << ts("Recent #{item_name.pluralize}")
     elsif collection.total_pages < 2
-      header = pluralize(collection.size, item_name)
+      header << pluralize(collection.size, item_name)
     else
       total_entries = collection.total_entries
       total_entries = collection.unlimited_total_entries if collection.respond_to?(:unlimited_total_entries)
-      header = %{ %d - %d of %d }% [
-                collection.offset + 1,
-                collection.offset + collection.length,
-                total_entries
-                ] + item_name.pluralize
+      header << ts(" %{start_number} - %{end_number} of %{total} %{things}",
+                   start_number: collection.offset + 1,
+                   end_number: collection.offset + collection.length,
+                   total: total_entries,
+                   things: item_name.pluralize)
     end
-    if search.present? && search.query.present?
-      header << " found"
-    end
-    if parent
-      parent_text = case parent.class.to_s
-                    when 'Collection'
-                      " in #{link_to(parent.title, parent)}"
-                    when 'Pseud'
-                      " by #{parent.byline}"
-                    when 'User'
-                      " by #{parent.login}"
-                    end
-      if parent.is_a?(Tag)
-        parent_text = " in #{link_to_tag_with_text(parent, parent.name)}"
-      end
-      if @fandom.present?
-        parent_text << " in #{link_to_tag(@fandom)}"
-      end
-      header << parent_text
+    header << ts("found") if search.present? && search.query.present?
+
+    case parent
+    when Collection
+      header << ts("in %{collection_link}", collection_link: link_to(parent.title, parent))
+    when Pseud
+      header << ts("by %{byline}", byline: parent.byline)
+    when User
+      header << ts("by %{username}", username: parent.login)
     end
-    header.html_safe
+
+    header << ts("in %{tag_link}", tag_link: link_to_tag_with_text(parent, parent.name)) if parent.is_a?(Tag)
+    header << ts("in %{fandom_link}", fandom_link: link_to_tag(@fandom)) if @fandom.present?
```

Comment:
I'd _love_ to, but I don't have enough context for how this method arrives at the point at having a `@fandom` instantiated in the first place, whether it's the same as the `parent` entity, and what we could use instead of an instance variable. :/

Reply:
I have figured out where this is used! It's if you follow a fandom link on someone's dashboard, which will take you to a URL like /users/sarken/works?fandom_id=164274.

However, I don't think it's super necessary to do anything about this here.